### ISSUE TITLE
Dashboard for Princeton guest accounts

### ISSUE THREAD

**Issue by hectorcorrea** [pulibrary/pdc_describe#993]
The dashboard page is broken for Princeton guest accounts. Users get a generic error page when accessing `https://pdc-describe-staging.princeton.edu/describe/users/guest-user@gmail.com`. 

The log file shows the following: 
```
ActiveRecord::RecordNotFound (can't find record with friendly id: "guest-user@gmail")
```

The user can navigate to the Works list page `https://pdc-describe-staging.princeton.edu/describe/works` and select a work from there.

**Comment by escowles** [comment]
@hectorcorrea This problem rings a bell — I feel like we had another app with a similar problem with the ".com" or other domain getting interpreted as a format suffix. Does it work if you URL-encode the address to `guest-user@gmail%2Ecom`?

**Comment by jrgriffiniii** [comment]
I suspect that the application is erroneously adding `@princeton.edu` to e-mail addresses when e-mail addresses already feature a different domain name for the user.

When testing this error within the `staging` environment, I found the following:

```bash
irb(main):004:0> User.find_by(email: "ppplpdctest@gmail.com")
=> nil
irb(main):005:0> User.find_by(email: "ppplpdctest@gmail.com@princeton.edu")
=> #<User id: 150, email: "ppplpdctest@gmail.com@princeton.edu", created_at: "2023-02-17 13:32:55.903253000 -0500", updated_at: "2023-02-17 13:32:55.903253000 -0500", provider: "cas", uid: "ppplpdctest@gmail.com", orcid: nil, default_collection_id: 1, display_name: "Leslie", full_name: "Leslie Smith", family_name: "Smith", email_messages_enabled: true>
```

**Comment by hectorcorrea** [comment]
The code where the error is triggered seems to be here: https://github.com/pulibrary/pdc_describe/blob/main/app/controllers/users_controller.rb#L55-L58

```
def set_user
  @user = User.friendly.find(params[:id])
  redirect_to action: action_name, id: @user.friendly_id, status: :moved_permanently unless @user.friendly_id == params[:id]
end
```

**Comment by hectorcorrea** [comment]
The issue in this particular case was quite surprising: We are sending the user to "/users/somebody@gmail.com" but Rails is interpreting the ".com" as a file extension and splitting the value in the URL into `id=somebody@gmail` and `format=com` and that's what's causing the look up by email "somebody@gmail" to fail. 

I did a quick fix in PR #995 to handle this.

**Comment by hectorcorrea** [comment]
@escowles you were right, that was the issue.
